autogen/agentchat/conversable_agent.py

```diff
@@ -61,6 +62,9 @@ def __init__(
         llm_config: Optional[Union[Dict, Literal[False]]] = None,
         default_auto_reply: Optional[Union[str, Dict, None]] = "",
         description: Optional[str] = None,
+        use_agent_stream: Optional[bool] = False,
```

Comment:
I can easily imagine more than just two types of output stream because there are so many different ways to stream data on the internet. Why don't we define `AgentOutput` protocol for outputting data and implement two initial classes: `PrintAgentOutput` and `SocketAgentOutput` implementing the protocol? That way we would be extensible in future without being forced to change the `ConversableClient` class.

Reply:
+1 On the suggestion.

Reply:
In this approach, should we plan for mutually exclusive outputs or a list of outputs? E.g. both print and socket outputs happening.